src/modding/ModLoader.cs

```diff
@@ -0,0 +1,278 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Godot;
+using File = System.IO.File;
+using Path = System.IO.Path;
+
+/// <summary>
+///   Handles loading mods, and auto-loading mods, and also showing related errors etc. popups
+/// </summary>
+public class ModLoader : Node
+{
+    private static ModLoader instance;
+    private static ModInterface modInterface;
+
+    private readonly List<string> loadedMods = new();
+
+    private readonly Dictionary<string, IMod> loadedModAssemblies = new();
+
+    private bool firstExecute = true;
+
+    private ModLoader()
+    {
+        instance = this;
+
+        // The reason why mods aren't loaded here already is that this object can't be attached to the scene here yet
+        // so we delay mod loading until this has been attached to the main scene tree
+    }
+
+    public static ModLoader Instance => instance;
+
+    /// <summary>
+    ///   The mod interface the game uses to trigger events that mods can react to
+    /// </summary>
+    public static ModInterface ModInterface => modInterface;
+
+    /// <summary>
+    ///   Finds a mod and loads its info
+    /// </summary>
+    /// <param name="name">The internal (folder) name of the mod</param>
+    /// <param name="failureIsError">If true, failure to find a mod is printed out</param>
+    /// <returns>The mod details if the mod could be loaded</returns>
+    public static FullModDetails LoadModInfo(string name, bool failureIsError = true)
+    {
+        using var currentDirectory = new Directory();
+
+        foreach (var location in Constants.ModLocations)
+        {
+            var modsFolder = Path.Combine(location, name);
+
+            if (!currentDirectory.DirExists(modsFolder))
+                continue;
+
+            if (currentDirectory.FileExists(Path.Combine(modsFolder, Constants.MOD_INFO_FILE_NAME)))
+            {
+                var info = ModManager.LoadModInfo(modsFolder);
+
+                if (info == null)
+                {
+                    GD.PrintErr("Failed to load info for mod \"", name, "\" from: ", modsFolder);
+                    return null;
+                }
+
+                return new FullModDetails(name) { Folder = modsFolder, Info = info };
+            }
+        }
+
+        if (failureIsError)
+            GD.PrintErr("No folder found for mod: ", name);
+        return null;
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
```

Comment:
Those base calls are not necessary

Reply:
For error-proofing reasons I now prefer to leave them. We've had 2 bugs so far from missed base calls, and I don't want anymore.